src/site/stages/build/drupal/api.js

```diff
@@ -15,7 +15,30 @@ function encodeCredentials({ username, password }) {
 }
 
 function getDrupalClient(buildOptions) {
-  const { address, credentials } = DRUPALS[buildOptions.buildtype];
+  let drupalConfig = {};
+
+  const drupalBuildOptions = {
+    address: buildOptions['drupal-address'],
+    credentials: {
+      user: buildOptions['drupal-user'],
+      password: buildOptions['drupal-password'],
+    },
+  };
+
+  if (buildOptions.buildtype === ENVIRONMENTS.LOCALHOST) {
+    // On localhost, build args should take priority.
+    drupalConfig = {
+      ...DRUPALS[ENVIRONMENTS.VAGOVDEV],
+      drupalBuildOptions,
+    };
+  } else {
+    drupalConfig = {
+      drupalBuildOptions,
+      ...DRUPALS[buildOptions.buildtype],
+    };
+  }
+
+  const { address, credentials } = drupalConfig;
```

Comment:
The idea here is that -

1. On localhost, you can pass your own Drupal configuration via the command line or environment variables, but the config defaults to Dev otherwise. 
2. The prod config omits the `credentials` property, so it has to be passed as either an env variable or command arg.